[CLOUD-1382] [CLOUD-1383] Add openshift.io/display-name to all image stream tag definitions

kyguy opened the pull request:
…s to all image stream tag definitions

Some files have multiple image Stream Tag definitions, I added the openshift/display-name var to the image stream tag definitions which refer to the application being built from the template. For example, for templates like eap64-amq-persistent-s2i.json, I added the openshift/display-name vars to the image stream tag definitions which were named ${APPLICATION_NAME}:latest but not to the ones named "jboss-eap64-openshift:1.4" or "jboss-amq-62:1.3". Is this the right way to do this? 

*NOTE: My script cleaned up some typos and indentation(a couple of the datavirt files had duplicate env lists). I can put this in a seperate commit/PR if you think that is better.

rcernich commented:
Hey @kyguy, sorry it took me so long to get to this.  Wrt the extraneous changes, please remove them (or use OrderedDict in your script, assuming it's py).  They pollute the diff's with unrelated changes.

Regarding the display names, these should apply to the items in jboss-image-streams.json, not in the templates.

rcernich commented:
Since you're going to be in there, feel free to address CLOUD-1382 at the same time.

kyguy commented:
- Put CLOUD-1382 and CLOUD-1383 changes together and updated the correct files. Let me know if the labels look appropriate

- I put the indentation and spacing fixes in a separate commit as part of this MR, is that proper protocol? Should I put the format changes in a separate MR? I have some ideas about future format handling below, let me know what you think:

I have been using a Python script with the json library to make changes across the template files.

In order to keep the diffs clean I have been using the following arguments when parsing and writing to the template files.
data = OrderedDict() ... json.dump(data, f, indent=4, separators=(',', ': '))

- The script corrects any inconsistencies among the files(indentations, spacing). It seems that my script occasionally comes across these inconsistencies and auto corrects them. For the sake of keeping the templates clean, I was thinking we declare a format standard for the template json files. For instance, we could declare that a proper indentation is 4 spaces and that there be a single space after separators(such as colons). What do you think?

- If you think this is a good idea, I can add a cleaning script that people can run before they commit their changes. By doing this, we can reduce the formatting commits and keep the formatting of the files consistent.

kyguy commented:
I mimicked the format of the OpenShift example image-streams. Although the OpenShift example image-streams have the "openshift.io/display-name" label in both the annotations and tags, they don't seem to appear in the latest version of the OpenShift web console for the image-streams.  Maybe the "openshift.io/display-name" values (for the image-streams) are planned for a future release? Still investigating this though, I will let you know what I find

rcernich commented:
Not sure what you mean.  Looking at the Ruby image, the display-name is "Ruby" while the name is ruby.  We can't change the name of the image streams, as this is what is used to reference the image stream in the rest of OpenShift.  Looking at the console, it displays "Ruby" not "ruby."

It also appears that the description display is specific to the tag selected in the drop down.

kyguy commented:
Updated the with metadata annotations, so now the openshift.io/display-names are appearing in the catalogue, I also updated the openshift.io/display-names to match the descriptions better
